sys/ssp/ssp.c

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_ssp Stack Smashing Protector
+ * @ingroup     sys
+ * @{
+ *
+ * @file
+ * @brief       Stack Smashing Protector (SSP) helper functions
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+
+#include "panic.h"
+
+#if UINT32_MAX == UINTPTR_MAX
+#define STACK_CHK_GUARD 0xe2dee396
+#else
+#define STACK_CHK_GUARD 0x595e9fbd94fda766
+#endif
```

Comment:
`UINTPTR_MAX` may be smaller than `UINT32_MAX` in practice and might be bigger (or smaller) than `UINT64_MAX` in theory (theory wrt to our target platform choice) - so at least put a warning there in case it is.